src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseApprovalService.cs

```diff
@@ -188,6 +195,110 @@ public class ReleaseApprovalService : IReleaseApprovalService
+
+            // The cron expressions are configurable per environment to allow different schedules for testing.
+
+            // There's a requirement that (1) and (2) always run at specific times in the Prod environment,
+            // e.g. (1) runs at 00:00:00 and (2) runs at 09:30:00 irrespective of daylight saving time.
+            // To avoid needing to adjust a cron expression for daylight saving time twice a year to express
+            // the desired time in UTC (e.g. changing it from '0 30 9 * * *' to '0 30 8 * * *' in British Summer Time),
+            // the functions have been configured to run in the UK timezone rather than the UTC default.
+
+            // Check if the functions have a scheduled occurrence on the publish date in the range between midnight
+            // (inclusive) and midnight the following day (exclusive), UK time.
+
+            var ukTimeZone = DateTimeExtensions.GetUkTimeZone();
+            var fromUtc = TimeZoneInfo.ConvertTimeToUtc(publishDate.Date, ukTimeZone);
+            var toUtc = fromUtc.AddDays(1).AddTicks(-1);
+
+            // The publish date cannot be scheduled if it's already passed
+            if (_dateTimeProvider.UtcNow > toUtc)
+            {
+                return false;
+            }
+
+            // The range should begin now rather than at midnight if the publish date is today
+            if (_dateTimeProvider.UtcNow > fromUtc)
+            {
+                fromUtc = _dateTimeProvider.UtcNow; 
+            }
+
+            // Publishing won't occur unless there's an occurrence of (1) between the publishing range
+            if (TryGetNextOccurrenceForCronExpression(
+                    cronExpression: _options.PublishReleasesCronSchedule,
+                    fromUtc: fromUtc,
+                    toUtc: toUtc,
+                    timeZoneInfo: ukTimeZone,
+                    fromInclusive: true,
+                    toInclusive: true,
+                    out var nextOccurrenceUtc))
+            {
+                // Publishing won't occur unless there's an occurrence of (2) after (1) but before the end of the range
+                return TryGetNextOccurrenceForCronExpression(
+                    cronExpression: _options.PublishReleaseContentCronSchedule,
+                    fromUtc: nextOccurrenceUtc.Value,
+                    toUtc: toUtc,
+                    timeZoneInfo: ukTimeZone,
+                    fromInclusive: true,
+                    toInclusive: true,
+                    out _);
+            }
+
+            return false;
+        }
+
+        private static bool TryGetNextOccurrenceForCronExpression(string cronExpression,
+            DateTime fromUtc,
+            DateTime toUtc,
+            TimeZoneInfo timeZoneInfo,
+            bool fromInclusive,
+            bool toInclusive,
+            [NotNullWhen(true)] out DateTime? nextOccurrenceUtc)
```

Comment:
The `out` variable of type `DateTime?` in the TryGet pattern will be null when the method returns false, i.e. when there's no next occurrence.

It's a code analysis attribute to tell the compiler the variable won't be null when the method returns true.

It prevents the warning "Nullable value type can be null" in the example below without it.

```
if (TryGetNextOccurrenceForCronExpression(..., out var nextOccurrenceUtc))
{
    var dateValue = nextOccurrenceUtc.Value;
}
```

which would otherwise need to be supressed as

```
if (TryGetNextOccurrenceForCronExpression(..., out var nextOccurrenceUtc))
{
    var dateValue = nextOccurrenceUtc!.Value;
}
```

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis